scraper.py

```diff
@@ -0,0 +1,118 @@
+    )
+
+def task_scrape():
+    with open("config.json") as file:
+        data=json.load(file)
+        service=Service(data["path"])
+        driver = webdriver.Chrome(service=service)
+        jobs={"roles":[],
+            "companies":[],
+            "locations":[],
+            "experience":[],
+            "skills":[],
+            "dates":[],
+            "scraper_run_dt_time":[],
+            "jd_url":[]
+            }
+        final_df=pd.DataFrame(jobs)    
+        #This outer loop is for number of pages to be scraped
+        for role in data["job_roles"]:
+            jobs={key:[] for key in jobs}
+            #print("hello",jobs)
+            for i in range(1,3):
+                driver.get("https://www.naukri.com/{}-jobs-{}".format(role,i))
+                time.sleep(4)
+                job_container = driver.find_elements(By.CSS_SELECTOR,".jobTuple.bgWhite.br4.mb-8")
+                # scraping the details from webpage
+                for job in job_container:
+                    driver.implicitly_wait(20)
+                    role=job.find_element(By.CSS_SELECTOR,"a.title.fw500.ellipsis").text
+                    company=job.find_element(By.CSS_SELECTOR,"a.subTitle.ellipsis.fleft").text
+                    location=job.find_element(By.CSS_SELECTOR,".fleft.grey-text.br2.placeHolderLi.location").text
+                    try:
+                        exp=job.find_element(By.CSS_SELECTOR,".fleft.grey-text.br2.placeHolderLi.experience").text
+                    except Exception:
+                        exp="0 yrs"      
+                    skills=job.find_element(By.CSS_SELECTOR,".tags.has-description").text
+                    date_string=job.find_element(By.CSS_SELECTOR,"[class^='type br2 fleft']").text  
+                    # date_string contains strings like 2 day ago,just now,few hours ago                                                                                
+                    jd=job.find_element(By.TAG_NAME,"a").get_attribute("href")
+                    date=re.findall(r'\d+',date_string) #extracting numbers out of the date_string
+                    jobs["roles"].append(role)
+                    jobs["companies"].append(company)
+                    jobs["locations"].append(location)
+                    jobs["experience"].append(exp)
+                    jobs["skills"].append(skills)
+                    jobs["dates"].append(date)
+                    jobs["jd_url"].append(jd)
+                    jobs["scraper_run_dt_time"].append(datetime.today())                   
+            jobs["dates"]=replacing_blank_dates_and_finding_date_from_days_count(jobs["dates"])
+            try:
+                print(type(jobs))
+                dataframe=dictionary_to_df_transformation(jobs)
+                print(type(dataframe))
+                dataframe[15:25]
+                final_df=final_df.append(dataframe)
+            except:
+                logging.error("Error in dict_to_df")                
```

Comment:
add the actual error captured Except Exception as err : 
logging.error("Error captured in dict to df tranformer "+str(err))